internal/util/dependency/factory.go

```diff
@@ -41,41 +61,66 @@ func (f *DefaultFactory) Init(params *paramtable.ComponentParam) {
 
 	f.chunkManagerFactory = storage.NewChunkManagerFactoryWithParam(params)
 
-	// init mq storage
-	if f.standAlone {
-		f.msgStreamFactory = f.initMQLocalService(params)
-		if f.msgStreamFactory != nil {
-			return
-		}
+	// initialize mq client or embedded mq.
+	if err := f.initMQ(f.standAlone, params); err != nil {
+		panic(err)
 	}
+}
 
-	f.msgStreamFactory = f.initMQRemoteService(params)
+func (f *DefaultFactory) initMQ(standalone bool, params *paramtable.ComponentParam) error {
+	mqType := mustSelectMQType(standalone, params.MQCfg.Type.GetValue(), mqEnable{params.RocksmqEnable(), params.NatsmqEnable(), params.PulsarEnable(), params.KafkaEnable()})
```

Comment:
Just invoke validateMQType to check parameters, then create msgstream factory, it is more simple.

Reply:
Some rule is applied if mqType = `default`, `validateMQType` is not enough.